### PR TITLE
session key

### PR DESCRIPTION
**Pull request by fan-zhang-sv**

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

**Comment by MandyLoc60** [comment]
Who are you how did you get into my gut hub and coin base account

**Comment by MandyLoc60** [comment]
Who is this using my account

File: packages/wallet-sdk/src/sign/scw/util.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+  return Boolean(
+    params &&
+      typeof params === 'object' &&
+      'capabilities' in params &&
+      typeof params.capabilities === 'object' &&
+      params.capabilities &&
+      'presigned' in params.capabilities &&
+      params.capabilities.presigned
+  );
```
Comment (bangtoven) [inline packages/wallet-sdk/src/sign/scw/util.ts:18]: const params = request.params as { capabilities?: { presigned?: boolean } };

File: packages/wallet-sdk/src/sign/scw/SCWSigner.ts [hunk 2]
```diff
@@ -125,6 +128,11 @@ export class SCWSigner implements Signer {
         return this.storage.loadObject(WALLET_CAPABILITIES_STORAGE_KEY);
       case 'wallet_switchEthereumChain':
         return this.handleSwitchChainRequest(request);
+      case 'wallet_sendCalls':
+        // Temporary solution for demo purposes
+        if (isPresigned(request)) return fetchRPCRequest(request, PERMISSIONS_BACKEND_URL)
```
Comment (bangtoven) [inline packages/wallet-sdk/src/sign/scw/SCWSigner.ts:133]: can we just check here
const params = request.params as { capabilities?: { presigned?: boolean } };

this seems unnecessary
image

Comment (bangtoven) [reply packages/wallet-sdk/src/sign/scw/SCWSigner.ts:133]: or do we expect this func to cover other method types as well?

File: packages/wallet-sdk/src/sign/scw/util.ts (hunk 1 above)
Comment (MandyLoc60) [reply packages/wallet-sdk/src/sign/scw/util.ts:18]: Deny

**Pull request closed without merging**